magic/fetcher.py

```diff
@@ -212,6 +212,61 @@ def get_frontside(scr_card: Dict) -> str:
     result_cardnames = [get_frontside(obj) for obj in result_data]
     return total_cards, result_cardnames
 
+async def dreadrise_count_cards(query: str) -> Tuple[int, Optional[str]]:
+    """
+        If the search succeeds, returns the number of results.
+        If it doesn't, returns -1 and the error given by the engine.
+    """
+    domain = configuration.get_str('dreadrise_url')
+    query = fetch_tools.escape(query)
+    url = f'{domain}/cards/find?q={query}+output:resultcount'
+    count_txt = await fetch_tools.fetch_async(url)
+    try:
+        return int(count_txt), None  # if this fails, the query errored
+    except ValueError:
+        return -1, count_txt
+
+async def dreadrise_search_cards(query: str, page_size: int = 60, pd_mode: Literal[1, 0, -1] = 0) -> List[str]:
```

Comment:
It might be better to represent the tristate `pd_mode` as `{True,False,None}`? Not important.